### DISCUSSION TITLE
Some Feature Requests

### DISCUSSION THREAD

**Discussion by SakiSakiSakiSakiSaki** [Decimation/SmartImage#35] [Ideas]
I appreciate the work you've put into this tool, @Decimation.

---
I have a particular use-case where I'm replacing images on my drive for their source-variants, and downloading them via other tools I have (Using Twitter's API and a Pixiv tool). Part of replacing, means deleting the version I have on my drive that I'm using to search with, so I just had a couple of requests.

- Ability to queue multiple images, and have it cycle between them.
    - Adding onto this, there could be an option to automatically search an input without having to hit the "run" button every time, both for single image searches and batch queued image searches. 

- Option to open the "raw" search results page of a particular engine, instead of the highest quality variant. For example, automatically opening the SauceNao and Ascii2D search result pages, and skimming the options my self. 

- A "Delete Image" button, so I don't have to go back and forth between your program and File Explorer to delete the image and then continue with the next image.

---
Let me know the feasibility of this, I wouldn't mind passing some gratitude along via a donation :)

**Comment by Decimation** [comment]
Thanks for the compliments! I will see about implementing the features you describe in the next release. As for the ability to open raw results, you should be able to do this already.

**Reply by SakiSakiSakiSakiSaki** [reply]
> Thanks for the compliments! I will see about implementing the features you describe in the next release. As for the ability to open raw results, you should be able to do this already.

Sorry I should've specified. I meant the ability to **automatically** open the raw results page, similar to how you can automatically opening the highest quality variant; being able to set it on the Priority Engines tab or something would make the most sense.

Or maybe I'm mistaken, and there's already a Config toggle to have the SauceNAO and Ascii2D search results page open in my browser, or maybe I'm using the word "raw" incorrectly here.

**Comment by Decimation** [comment]
Try the [3.0.1 pre-release](https://github.com/Decimation/SmartImage/releases/tag/v3.0.1-pre). I added the delete functionality and the `Open raw` configuration option. 

- The `Open raw` feature behaves as you requested (toggle it in the `Config` dialog)

- I designed the delete functionality such that it doesn't ask for confirmation (for efficiency), and to prevent against accidental deletions, it sends the file to the recycle bin.

The queue feature will require a bit more work but I'll try to implement it soon.

Additionally, I'm working on a command-line version of the program (bypassing the shell UI entirely) which you may find useful (once it's complete).

**Comment by SakiSakiSakiSakiSaki** [comment]
This is a great start! I really appreciate you going out of the way for my request. There are a few kinks as expected since you just cooked this up right now, but I really appreciate this!

**Reply by Decimation** [reply]
Thank you for the donation! I will implement your feature requests ASAP (next release).

**Comment by Decimation** [comment]
How should the queue behave specifically? 

> Ability to queue multiple images, and have it cycle between them.

>>    Adding onto this, there could be an option to automatically search an input without having to hit the "run" button every time, both for single image searches and batch queued image searches.

Do you want to be able to tab between multiple results, and searches in the queue run concurrently in the background?

**Comment by SakiSakiSakiSakiSaki** [comment]
So what I was imagining was the ability to select multiple images and drag them in all at once, and it adds to a "queue". You'd also be able to drop single images into the window, with it being appended to the end of the queue.

I'm not sure what the feasibility of having a tab like system on CML could look like, and having multiple concurrent searches may result in being rate-limited on one of the **many** search engines you have provided. For example, SauceNao only allows 3 searches within a 30 second window for unregistered users.

For simplicity's sake, the easiest implementation would likely be having them search consecutively. After you get your result, check out the source image, delete the image you fed the program, or whatever, you should be able to toggle to the next image in the queue with a designated "Next" button. This "Next" button would overwrite/flush the results currently list, and begin a new search. A `place_in_queue/queue_size` counter would be great to have.

**Comment by SakiSakiSakiSakiSaki** [comment]
@Decimation Tacking onto my previous written requests, I think it would be ideal if there was a toggle to disable the "sound effect" for a successful search.
